SharedPluginFiles/FormItPluginUI.js

```diff
@@ -2,6 +2,45 @@
 window.FormIt = window.FormIt || {};
 FormIt.PluginUI = FormIt.PluginUI || {};
 
+/// UI utilities ///
+
+// converts a numeric value to a properly-formatted FormIt dimension string
+FormIt.PluginUI.convertValueToDimensionString = function(value, callbackFunction)
+{
+    // if the value is not a number, it could be a dimension string already
+    // so try to convert it to a number
+    if (isNaN(Number(value)))
+    {
+        window.FormItInterface.CallMethod("FormIt.StringConversion.StringToLinearValue", value, function(result)
+        {
+            // parse the result
+            result = JSON.parse(result);
+
+            // if the API indicates it was able to convert the dimension to a value, return it
+            if (result.first === true)
+            {
+                window.FormItInterface.CallMethod("FormIt.StringConversion.LinearValueToString", Number(result.second), callbackFunction);
+            }
+            // otherwise, the value was not a valid number or dimension
+            // so return a single unit of value
+            else
+            {
+                window.FormItInterface.CallMethod("FormIt.StringConversion.LinearValueToString", 1, callbackFunction);
```

Comment:
This case, as mentioned in the comments, is for when the value is just gibberish - not a number, and not a dimension. So I return 1 to avoid bad things happening with invalid inputs.

Reply:
But could the user accidentally moved the cursor and then loses what they just typed in that might not have been complete?

Reply:
Sure, in which case this "1" code won't get hit. If it's a value, it goes into a different logic path, and if it's a partial dimension string, then I pass it to the FormIt "ConvertDimensionStringToValue" to try converting it first, and if that fails, only then do we reset to "1" - I'm basically doing what FormIt does, in that if you enter garbage for a dimension, it gets reset to 1 - like in the Material Editor dialog.